nodejs/eks/servicerole.ts

```diff
@@ -45,15 +45,22 @@ export interface ServiceRoleArgs {
      * Tags to apply to the role.
      */
     readonly tags?: pulumi.Input<{ [key: string]: pulumi.Input<string> }>;
+
+    /**
+     * The actions to allow in the assume role policy. Defaults to ["sts:AssumeRole"].
+     */
+    readonly assumeRoleActions?: pulumi.Input<string[]>;
 }
 
 /**
  * The ServiceRole component creates an IAM role for a particular service and attaches to it a list of well-known
  * managed policies.
  */
 export class ServiceRole extends pulumi.ComponentResource {
-    // The service role.
-    public readonly role: pulumi.Output<aws.iam.Role>;
```

Comment:
Previously this internal component exported a single output for the role, that depended on all policy attachments.
Because of that it becomes unknown when attaching new policies.

In order to upgrade an existing cluster to Auto Mode, you need to attach new policies. The Cluster's service role property triggers replacements, so the preview would've shown that the cluster gets replaced.
This new approach is a bit more careful. We expose am output for the fully resolved role (with all policies attached) and the direct role resource itself.
Now the consumers can pick and choose what they want.